### PR TITLE
Update genesis block retrieval configuration - Closes #640

### LINKED ISSUE #640: Update genesis block retrieval configuration
### Description

1. Create config section that keep information about the specific networks (testnet/mainnet).
2. Make usage of the FS JSON by default.

### Motivation

Lisk Service need to distinguish networks from each other to used fixed remote JSON files.

### PR DESCRIPTION
**Pull request by sameersubudhi**
### What was the problem?
This PR resolves #640 

### How was it solved?
- [x] Created config section specific to the networks (testnet/mainnet)
- [x] Preferential fetching of Genesis block from FS if exists
- [x] Access genesis block by `height` / `blockId`
- [x] Error handling for `peerCache.js` and `knownAccounts.js`
- [x] Avoid re-indexing already indexed Genesis block

### How was it tested?
Local

### CODE REVIEW

File: services/core/shared/core/compat/sdk_v5/accounts.js [hunk 1]
```diff
@@ -108,7 +108,10 @@ const getAccountsFromCore = async (params) => {
 };
 
 const indexAccountsbyAddress = async (addressesToIndex, isGenesisBlockAccount = false) => {
-	const { data: accountsToIndex } = await getAccountsFromCore({ addresses: addressesToIndex });
+	const { data: accountsToIndex } = await getAccountsFromCore({
+		addresses: addressesToIndex
+			.filter((v, i, a) => a.findIndex(t => (t === v)) === i), // Remove duplicates
```
Comment (MichalTuleja) [inline services/core/shared/core/compat/sdk_v5/accounts.js:113]: Please make a util function (ie. `dropDuplicates`).

File: services/core/shared/core/compat/sdk_v5/accounts.js [hunk 2]
```diff
@@ -224,14 +227,15 @@ const indexAccountsbyPublicKey = async (accountInfoArray) => {
 
 	const { data: accountsToIndex } = await getAccountsFromCore({
 		addresses: accountInfoArray
-			.map(accountInfo => getBase32AddressFromPublicKey(accountInfo.publicKey)),
+			.map(accountInfo => getHexAddressFromPublicKey(accountInfo.publicKey))
+			.filter((v, i, a) => a.findIndex(t => (t === v)) === i), // Remove duplicates
```
Comment (MichalTuleja) [inline services/core/shared/core/compat/sdk_v5/accounts.js:231]: Please make a util function (ie. `dropDuplicates`).

File: services/core/shared/core/compat/sdk_v5/blocksUtils.js [hunk 3]
```diff
@@ -23,25 +24,43 @@ const {
...
+
+const loadConfig = async () => {
+	// Direct invocation of action necessary to avoid circular dependency
+	const apiClient = await getApiClient();
+	const { networkIdentifier } = await apiClient.node.getNodeInfo();
+
+	const [networkConfig] = config.network.filter(c => c.identifier === networkIdentifier);
+	genesisBlockURL = networkConfig.genesisBlockUrl;
+
+	genesisBlockFilePath = `./shared/core/compat/sdk_v5/static/${networkConfig.name}/genesis_block.json`;
```
Comment (MichalTuleja) [inline services/core/shared/core/compat/sdk_v5/blocksUtils.js:53]: Please store that files in a separate dir `./data/${networkConfig.name}`. Please add the data path to gitignore and dockerignore.

File: services/core/shared/core/compat/sdk_v5/coreApi.js [hunk 4]
```diff
@@ -61,25 +76,26 @@ const getBlockByID = async id => {
...
+		const genesisBlockId = getGenesisBlockId();
+		if (ids.includes(genesisBlockId)) {
+			const remainingIds = ids.filter(id => id !== genesisBlockId);
+			const genesisBlockResult = await getBlockByID(genesisBlockId);
+			if (remainingIds.length) {
+				const { data: [genesisBlock] } = genesisBlockResult;
+				const { data: [...remainingBlocks] } = await getBlocksByIDs(remainingIds);
+				return { data: [genesisBlock, ...remainingBlocks] };
+			}
+			return genesisBlockResult;
```
Comment (MichalTuleja) [inline services/core/shared/core/compat/sdk_v5/coreApi.js:89]: Please avoid code duplication. The retrieval logic gets complex and all functions could call data with multiple block methods, either by IDs or range.

File: services/core/shared/core/delegates.js [hunk 5]
```diff
@@ -249,6 +249,34 @@ const reload = async () => {
 	await computeDelegateStatus();
 };
 
+// Keep the delegate cache up-to-date
+Signals.get('newBlock').add(async data => {
```
Comment (MichalTuleja) [inline services/core/shared/core/delegates.js:253]: It would be helpful to call the function by name, to indicate what it does.

**Review by MichalTuleja — CHANGES_REQUESTED** [review]

File: services/core/shared/core/compat/sdk_v5/coreApi.js (hunk 4 above)
Comment (sameersubudhi) [reply services/core/shared/core/compat/sdk_v5/coreApi.js:89]: `getBlocksByIDs` and `getBlocksByHeightBetween` utilise a similar approach but the implementation invoke their corresponding singular methods i.e. `getBlockByID` and `getBlocksByHeightBetween`.

File: services/core/shared/core/compat/sdk_v5/blocks.js [hunk 6]
```diff
@@ -353,30 +366,40 @@ const deleteBlock = async (block) => {
...
-		// eslint-disable-next-line no-await-in-loop
-		await indexAccountsbyAddress(
-			accountAddressesToIndex.slice(i * PAGE_SIZE, (i + 1) * PAGE_SIZE),
-			true,
-		);
-	}
+	const blocksDB = await getBlocksIndex();
+	const [indexedGenesisBlock] = await blocksDB.find({ height: genesisHeight });
+
+	if (indexedGenesisBlock !== undefined) {
```
Comment (nagdahimanshu) [inline services/core/shared/core/compat/sdk_v5/blocks.js:372]: This looks more clean:
```suggestion
	if (indexedGenesisBlock) {
```

**Review by nagdahimanshu — APPROVED** [review]
Looks good apart from one suggestion 👍

File: services/core/shared/core/compat/sdk_v5/coreApi.js (hunk 4 above)
Comment (MichalTuleja) [reply services/core/shared/core/compat/sdk_v5/coreApi.js:89]: It will be updated in #642

**Review by MichalTuleja — APPROVED** [review]

**Pull request merged**